geonode/harvesting/harvesters/geonode.py

```diff
@@ -442,24 +454,32 @@ def _get_resource_descriptor(
                 csw_record.xpath(
                     "gmd:dateStamp/gco:DateTime/text()", namespaces=csw_record.nsmap)[0],
             ).replace(tzinfo=dt.timezone.utc),
-            reference_systems=[f"EPSG:{crs_epsg_code}"],
-            identification=get_identification_descriptor(
-                csw_record.xpath("gmd:identificationInfo", namespaces=csw_record.nsmap)[0]
-            ),
+            reference_systems=[crs],
+            identification=identification_descriptor,
             distribution=self.get_distribution_info(
                 csw_record.xpath(
                     "gmd:distributionInfo", namespaces=csw_record.nsmap)[0],
                 api_record,
                 harvestable_resource,
+                identification_descriptor,
+                crs
             ),
             data_quality=get_xpath_value(csw_record, ".//gmd:dataQualityInfo//gmd:lineage"),
         )
+        if harvestable_resource.remote_resource_type == GeoNodeResourceType.LAYER.value:
+            layer_type = (
+                GeoNodeLayerType.RASTER if api_record.get("store_type") == "coverageStore" else GeoNodeLayerType.VECTOR
```

Comment:
**WARNING** : `api_record.get("store_type") == "coverageStore"` this is valid for _old_ GeoNode instances, it won't work anymore with the new ones. I'm afraid we will need to keep this duality though, in order to support the import from the old ones.

Reply:
I am aware of it, and that is why this harvester is called `GeonodeLegacyHarvester` - it is supposed to be used with old GeoNode instances. Given the rocky state of the old API I think it will be better to have a more modern harvester worker, in addition to this one.

Reply:
It seems the old API even has a bug where in the list page the attribute is called `store_type` but in the detail page it is called `storeType`

Reply:
I'm leaving it like this, since I envision creating a `GeonodeHarvester` class that would explicitly only support the newer GeoNode API and have a cleaner implementation. Is it OK for you @afabiani?

Reply:
I don't care much about the legacy API. We can fix is or not. 
API v2 should have instead snake case properties everywhere though.

I'm with @ricardogsilva regarding the enumerations. I think `GeoNodeLayerType.RASTER` is better.